File: packages/es-markdown-core/src/model/heading.ts
import type { MarkupBase } from "./base.js";
import type { Markup } from "./markup.js";

/** An ATX or setext heading; the node does not record which form was written. */
export interface Heading extends MarkupBase<"heading"> {
    /** The heading level, 1 through 6. */
    readonly level: number;
    /** The heading's inline content in source order. */
    readonly content: readonly Markup[];
}

File: packages/kotlin-markdown-core/src/commonMain/kotlin/com/nouprax/markdown/core/model/Heading.kt
package com.nouprax.markdown.core

/** An ATX or setext heading. */
public class Heading internal constructor(
    override val id: MarkupID,
    override val revision: ULong,
    override val scope: Scope,
    /** The heading level, 1 through 6. */
    public val level: Int,
    /** The heading's inline content in source order. */
    public val content: kotlin.collections.List<Markup>,
) : Markup {
    override fun <Result> accept(visitor: MarkupVisitor<Result>): Result = visitor.visit(this)

    override fun equals(other: Any?): Boolean = markupEquals(this, other)

    override fun hashCode(): Int = markupHashCode(this)
}
